File: Example/ios/iOS Example/OptOutViewController.m
#import "OptOutViewController.h"
#import "HubjsTracker.h"


@interface OptOutViewController ()
@end


@implementation OptOutViewController


- (void)viewDidLoad {
  [super viewDidLoad];
  
  self.optOutSwitch.on = ![HubjsTracker sharedInstance].optOut;
  
}


- (IBAction)sendOptOutAction:(id)sender {
  
  UISwitch *option = (UISwitch*)sender;
  
  if (option.on) {
    
    [HubjsTracker sharedInstance].optOut = NO;
    
  } else {
    
    [HubjsTracker sharedInstance].optOut = YES;
    
  }
  
}


@end

File: Example/ios/iOS Example/OptOutViewController.swift
import UIKit
import HubjsTracker

class OptOutViewController: UIViewController {
    
    @IBOutlet weak var optOutSwitch: UISwitch!
    @IBAction func optOutSwitchChanged(_ sender: Any) {
        HubjsTracker.shared.isOptedOut = optOutSwitch.isOn
    }
    
    override func viewDidLoad() {
        super.viewDidLoad()
        optOutSwitch.isOn = HubjsTracker.shared.isOptedOut
    }
    
    override func viewDidAppear(_ animated: Bool) {
        super.viewDidAppear(animated)
        HubjsTracker.shared.track(view: ["menu","opt out"])
    }
}
